crypto/dilithium/p_dilithium_test.cc

```diff
@@ -679,70 +679,95 @@ TEST(Dilithium3Test, SIGOperations) {
-  std::vector<uint8_t> seed, msg, pk, sk, sm;
-
-  ASSERT_TRUE(t->GetAttribute(&count, "count"));
-  ASSERT_TRUE(t->GetBytes(&seed, "seed"));
-  ASSERT_TRUE(t->GetAttribute(&mlen, "mlen"));
-  ASSERT_TRUE(t->GetBytes(&msg, "msg"));
-  ASSERT_TRUE(t->GetBytes(&pk, "pk"));
-  ASSERT_TRUE(t->GetBytes(&sk, "sk"));
-  ASSERT_TRUE(t->GetAttribute(&smlen, "smlen"));
-  ASSERT_TRUE(t->GetBytes(&sm, "sm"));
-
-  // The KAT files generated by the dilithium team use the optional APIs that
-  // create a signature for a message m and append the message to the end of
-  // the signature. We only want to bring the APIs that create and verify just
-  // the signature, therefore each signature is a constant
-  // DILITHIUM3_SIGNATURE_BYTES and we truncate the signed message down to a
-  // constant DILITHIUM3_SIGNATURE_BYTES.
-  uint8_t signature[DILITHIUM3_SIGNATURE_BYTES];
-  size_t signature_len = DILITHIUM3_SIGNATURE_BYTES;
-  sm.resize(signature_len);
-
-  // Convert string read from KAT to int
-  size_t mlen_int = std::stoi(mlen);
-
-  // Here we fix the DRBG (AES-CTR) so that we are able to seed it with the
-  // seed from the KAT (testing only)
-  pq_custom_randombytes_use_deterministic_for_testing();
-  pq_custom_randombytes_init_for_testing(seed.data());
-
-  // Generate our dilithium public and private key pair
-  EVP_PKEY_CTX *dilithium_pkey_ctx = EVP_PKEY_CTX_new_id(EVP_PKEY_DILITHIUM3, nullptr);
-  ASSERT_NE(dilithium_pkey_ctx, nullptr);
-
-  EVP_PKEY *dilithium_pkey = EVP_PKEY_new();
-  ASSERT_NE(dilithium_pkey, nullptr);
-
-  EXPECT_TRUE(EVP_PKEY_keygen_init(dilithium_pkey_ctx));
-  EXPECT_TRUE(EVP_PKEY_keygen(dilithium_pkey_ctx, &dilithium_pkey));
-  const DILITHIUM3_KEY *dilithium3Key = (DILITHIUM3_KEY *)(dilithium_pkey->pkey.ptr);
-  EXPECT_EQ(Bytes(pk), Bytes(dilithium3Key->pub, DILITHIUM3_PUBLIC_KEY_BYTES));
-  EXPECT_EQ(Bytes(sk), Bytes(dilithium3Key->priv, DILITHIUM3_PRIVATE_KEY_BYTES));
-
-  // Generate a signature for the message
-  bssl::ScopedEVP_MD_CTX md_ctx;
-  // We have to use EVP_DigestSign because dilithium supports the use of
-  // non-hash-then-sign (just like ed25519) so we first init EVP_DigestSign
-  // WITHOUT a hash function.
-  ASSERT_TRUE(EVP_DigestSignInit(md_ctx.get(), NULL, NULL, NULL, dilithium_pkey));
-  ASSERT_TRUE(EVP_DigestSign(md_ctx.get(), signature, &signature_len, msg.data(), mlen_int));
-  EXPECT_EQ(Bytes(sm), Bytes(signature, signature_len));
-
-  // Verify the signature for the message
-  ASSERT_TRUE(EVP_DigestVerify(md_ctx.get(), signature, signature_len, msg.data(), mlen_int));
+struct ML_DSA {
+  const char name[20];
+  const int nid;
+  const size_t public_key_len;
+  const size_t secret_key_len;
+  const size_t signature_len;
+  const char *kat_filename;
+};
 
-  EVP_PKEY_free(dilithium_pkey);
-  EVP_PKEY_CTX_free(dilithium_pkey_ctx);
-  md_ctx.Reset();
-}
+static const struct ML_DSA parameterSet[] = {
+  {"MLDSA67", NID_DILITHIUM3_R3, 1952, 4032, 3309,  "dilithium/kat/mldsa65.txt"},
```

Comment:
```suggestion
  {"MLDSA65", NID_DILITHIUM3_R3, 1952, 4032, 3309,  "dilithium/kat/mldsa65.txt"},
```

Reply:
Thanks, fixed in [ea859d1](https://github.com/aws/aws-lc/pull/1910/commits/ea859d1b4a527b801a531e415122f51475802817).